lib/dams/controller_helper.rb

```diff
@@ -898,28 +898,45 @@ def referrer_controller( request )
       end
     end
 
-  def metadata_display?(data)
-    result = false
-    return result unless data
-    data.each do |datum|
-      result = true if datum.include?('localDisplay') || datum.include?('metadataDisplay')
+    def metadata_display?(data)
+      result = false
+      return result unless data
+      data.each do |datum|
+        result = true if datum.include?('localDisplay') || datum.include?('metadataDisplay')
+      end
+      result
     end
-    result
-  end
 
     #---
     # Check to see if an object allows file download
     #
     # @return Boolean value
     #---
+    def can_download?(solr_doc, fileid = nil)
+      permissions = (Array(solr_doc['license_tesim']) + Array(solr_doc['otherRights_tesim'])).flatten.compact
+      return false if streaming_media_type?(solr_doc['resource_type_tesim']) && local_display?(permissions)
+      metadata_only_display = metadata_only_display?(permissions)
+      return !restricted_file?(fileid) if fileid && metadata_only_display
+      !metadata_only_display
+    end
 
-    def can_download?(document)
-      local_license = false
-      local_other_rights = document['otherRights_tesim'] && document['otherRights_tesim'].first.to_s.include?('localDisplay') ? true : false
-      if document['resource_type_tesim'] && document['resource_type_tesim'].first.to_s.include?('video')
-        local_license = document['license_tesim'] && document['license_tesim'].first.to_s.include?('localDisplay') ? true : false
-      end
-      !local_other_rights && !local_license
+    def restricted_file?(fileid)
+      fileid && fileid.include?('_2.jpg')
```

Comment:
What is specific to an `_2.jpg` being a restricted file? I don't think I understand the meaning behind this one.